configure.ac

```diff
@@ -106,13 +106,17 @@ else
     if test "x$enable_plugins" = xno; then
         AM_CONDITIONAL([BUILD_C_API], [false])
     elif test "x$enable_c_plugins" != xno; then
-        AC_CHECK_LIB([dl], [main],
-            [AM_CONDITIONAL([BUILD_C_API], [true]) LIBS="$LIBS -ldl" AC_DEFINE([HAVE_C], [1], [C support])],
-            [AS_IF(
-                [test "x$enable_c_plugins" = xyes],
-                    [AC_MSG_ERROR([dl library needed to run C plugins])],
-                [AM_CONDITIONAL([BUILD_C_API], [false])])
-            ])
+        # libdl doesn't exist as a separate library in OpenBSD/FreeBSD and is
```

Comment:
@arvedarved your BSD port doesn't seem to have this. What's your opinion on it?

Reply:
@pasis maybe you know too?

Reply:
I finally found https://svnweb.freebsd.org/ports/head/net-im/profanity/files/patch-configure.ac?revision=438599&view=markup (weirdly OpenBSD doesn't seem to have this?)

Reply:
https://svnweb.freebsd.org/ports/head/net-im/profanity/files/patch-configure.ac?revision=438599&view=markup

Reply:
Yes FreeBSD has dlopen in libc so does not need a "libdl". Maybe the configure should check if dlopen exists in either libc or libdl. There are several examples in other open source software.